torch/csrc/jit/codegen/cuda/executor.cpp

```diff
@@ -206,13 +207,16 @@ void FusionExecutor::compileFusion(
 
   const auto& kernel_summary = kernel->summary();
 
+  // We currently shouldn't allocate any more shared mem
+  //  tensors statically but could keep this path if
+  //  needed in later development.
```

Comment:
Please remove the static paths in a follow up. The other thing we may want to do is also get rid of the second expression evaluator in a cleanup PR since `kir::Kernel` inherits from `Fusion` now.

Reply:
Makes sense. Will do in a follow up. Thanks.